prusti-tests/tests/verify/pass/type-model/models-for-same-type-unsound.rs

```diff
@@ -0,0 +1,35 @@
+// This test instantiates two instances of `A` and checks that their models are independent.
+// Notice that `A` has no fields, which introduces a soundness issues.
+// The reason for this behavior is that the encoding of `A` (w/o fields) in Viper
+// assumes that all instances are equal, i.e. there is no notion of inequality for two `A`s.
+// Hence, we cannot refer to different models for different instances of `A`.
+// If `A` had fields, the code verifies as expected.
+//
+// This code will print a warning to a user.
```

Comment:
If this test is meant to check that the warning is printed, we can put it into `verify(_overflow)/ui`.

Reply:
There is a separate test which checks the warning in `verify/ui`.
This test is meant as a "documentation" of the behavior. It is maybe not needed, but I found it somewhat useful. Mentioning it in the user docs might be good enough though.

Reply:
Test deleted, other tests moved